src/figuresClasses.ts

```diff
@@ -1,19 +1,91 @@
-export interface Figure {
+enum Figures {
+  Triangle = 'triangle',
+  Circle = 'circle',
+  Rectangle = 'rectangle',
+}
+
+enum Colors {
+  Red = 'red',
+  Green = 'green',
+  Blue = 'blue',
+}
 
+export interface Figure {
+  shape: Figures,
+  color: Colors,
+  getArea():number;
 }
 
-export class Triangle {
+export class Triangle implements Figure {
+  shape: Figures = Figures.Triangle;
+
+  constructor(
+    public color: Colors,
+    public a: number,
+    public b: number,
+    public c: number,
+  ) {
+    if (a <= 0 || b <= 0 || c <= 0) {
+      throw new Error('Invalid side value of triangle');
+    }
 
+    if (a >= b + c || b >= a + c || c >= a + b) {
+      throw new Error('Sides cannot form a triangle');
+    }
```

Comment:
would be nice to create a method like `validateTriangle` or sth like that where you handle such cases

Reply:
Done.
Thank you!